apps/scan/frontend/test/helpers/mock_api_client.tsx

```diff
@@ -81,9 +72,7 @@ export function createApiMock() {
   const mockApiClient = createMockApiClient();
 
   function setAuthStatus(authStatus: InsertedSmartCardAuth.AuthStatus): void {
-    mockApiClient.getAuthStatus.mockImplementation(() =>
-      Promise.resolve(authStatus)
-    );
+    mockApiClient.getAuthStatus.expectRepeatedCallsWith().resolves(authStatus);
```

Comment:
If we like this approach, I will update the rest of the apps accordingly. Just wanted to get feedback first.

Reply:
Overall, I like the interface!

My main concern is just that the implementation of the interface is fairly hard to reason about. We shouldn't have to update the underlying implementation often, so it's not a big deal. But I am once again wondering whether it's the right call to reinvent the (Jest) wheel or not. I do still see the merits of strictly enforcing the call pattern, as brought up in this thread: https://github.com/votingworks/vxsuite/pull/3107#discussion_r1131736705. So I'm happy to continue down this line for now

Reply:
Yeah I agree it's hard to reason about! I puzzled a bit over ways to make it clearer.

I think the core difference between the Jest mock function and this mock is that Jest mocks differentiate the mock implementation from assertions about what calls were expected. Whereas with this mock, they are one and the same - the expected calls provide the "implementation" (return values). I think it's a pretty major difference that makes me feel like it's a new wheel, not necessarily a reinvention.

That being said, I'm still very open to this not being worthwhile. It's definitely a bit confusing having two different types of mock functions out there. (Though we did have that previously with fetch-mock anyway.)

I feel similar to you - down to keep experimenting with this while holding open the possibility that's it's not worth it.

Reply:
> Whereas with this mock, they are one and the same - the expected calls provide the "implementation" (return values). I think it's a pretty major difference that makes me feel like it's a new wheel, not necessarily a reinvention.

Thanks for that framing! Agree that that's more substantial than my earlier framing of just strictly enforcing the call pattern

> I feel similar to you - down to keep experimenting with this while holding open the possibility that's it's not worth it.

:100: